lldb/packages/Python/lldbsuite/test/lldbgdbproxy.py

```diff
@@ -0,0 +1,176 @@
+import logging
+import os
+import os.path
+import random
+
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.gdbclientutils import *
+import lldbgdbserverutils
+from lldbsuite.support import seven
+
+
+class GDBProxyTestBase(TestBase):
```

Comment:
nit: This class is pretty cool, however it would be nice if we could separate the proxy GDB server from the test base class, since one could want to use the proxy GDB server for something other than a test.

You could the mock class in lldb/example/python and add it to the lldb python module. This could be a follow-up.